### PR TITLE
305 Z-Index Map and Function

### PR DESCRIPTION
**Pull request by buttonwillowsix**
# READY FOR REVIEW

# Summary
- Making a z-index map and a function to easily access the map
- Obviously there is a lot more that could be done to extend this. The map provided here is based on existing elements with z-index set, but not with the same values (since 9999 should not be necessary here.)
- This PR does NOT include swapping out the existing values, but if this looks good, I could do that next.

# Needed By (Date)
- End of sprint would be nice 

# Urgency
- Not very

# Steps to Test

1. Use function z($level) somewhere and see it pull the correct value from the map or error out if you use a bad name

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: core/dist/css/decanter-grid.css [hunk 1]
```diff
@@ -1,1910 +0,0 @@
-html {
```
Comment (sherakama) [inline core/dist/css/decanter-grid.css]: I'm pretty sure we want to keep this file. Run and commit a new build.

File: package.json [hunk 2]
```diff
@@ -33,7 +33,8 @@
     "url": "https://github.com/SU-SWS/decanter/issues"
   },
   "dependencies": {
-    "bourbon": "^5.1.0"
+    "bourbon": "^5.1.0",
+    "webpack-cli": "^3.3.0"
```
Comment (sherakama) [inline package.json]: This isn't really a dependency.

File: package.json [hunk 3]
```diff
@@ -67,8 +68,6 @@
     "url-loader": "^1.1.2",
     "webpack": "^4.29.6",
     "webpack-assets-manifest": "^3.1.1",
-    "webpack-cli": "^3.3.0",
-    "webpack-command": "^0.4.2",
```
Comment (sherakama) [inline package.json]: Wathchew got against webpack-command?

File: core/src/scss/utilities/functions/z-index/_z-index.scss [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+//
+// A function to help us get the z-index from the level from the map
+//
+// $level - the name of the z-index level
+//
+// Style guide: Functions.z-index.level
+//
+
+
+@function z($level) {
```
Comment (sherakama) [inline core/src/scss/utilities/functions/z-index/_z-index.scss]: Seems straightforward enough. Do you plan on adding a highest or lowest function where they return +1 above the biggest number in $z-levels or -1 of the smallest?

File: core/src/scss/utilities/functions/z-index/_z-index.scss [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  @if map-has-key($z-levels, $level) {
```
Comment (sherakama) [inline core/src/scss/utilities/functions/z-index/_z-index.scss]: Perhaps make the `$level` param default to `default` as that is an option in the map.

**Comment by sherakama** [comment]
I think you are off to a good start. The details will be in how you implement the new map. Keep going.

File: core/dist/css/decanter-grid.css (hunk 1 above)
Comment (buttonwillowsix) [reply core/dist/css/decanter-grid.css]: Hm, seems like my stuff did more than I wanted it to. So yes.

File: package.json (hunk 3 above)
Comment (buttonwillowsix) [reply package.json]: It wouldn't run with webpack command. Let me try to fix!

File: core/src/scss/utilities/functions/z-index/_z-index.scss (hunk 4 above)
Comment (buttonwillowsix) [reply core/src/scss/utilities/functions/z-index/_z-index.scss]: Yes, but after looking at examples of that, I didn't see one that actually seemed to work.

**Pull request closed without merging**